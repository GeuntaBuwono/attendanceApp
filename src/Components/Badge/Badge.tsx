/* eslint-disable react-native/no-inline-styles */
import Label from 'Components/Text/Text'
import { View } from 'react-native'
import styled from 'styled-components'
import Colors from 'Styles/colors'

import { BadgeColorType, BadgeInterface, BadgeStyleInterface } from './Badge.type'

const badgeColorMapper: Record<BadgeColorType, string> = {
  grey: Colors.silver,
  topaz: Colors.topaz,
  red: Colors.redPink,
}

const BadgeStyled = styled(View)<BadgeStyleInterface>`
  background-color: ${(props) => badgeColorMapper[props.color]};
  border-radius: 6px;
  padding: 5px 10px;
`

const Badge = (props: BadgeInterface) => {
  const { label, color } = props
  return (
    <BadgeStyled color={color}>
      <Label restStyle={{ color: Colors.white, textAlign: 'center', fontWeight: 'bold' }}>
        {label}
      </Label>
    </BadgeStyled>
  )
}

export default Badge
