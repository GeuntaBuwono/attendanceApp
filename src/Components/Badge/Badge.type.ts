export type BadgeColorType = 'topaz' | 'grey' | 'red'
export type BadgeSizeType = 'small' | 'medium' | 'large'

export interface BadgeStyleInterface {
  color: BadgeColorType
  size?: BadgeSizeType
  variant?: 'capsule' | 'outline'
}
export interface BadgeInterface extends BadgeStyleInterface {
  label: string
}
