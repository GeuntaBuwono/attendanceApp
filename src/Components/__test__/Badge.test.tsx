/**
 * @format
 */

import 'react-native'

import { render } from '@testing-library/react-native'
import Badge from 'Components/Badge/Badge'
import React from 'react'
// Note: test renderer must be required after react-native.

describe('Badge', () => {
  it('Render Correctly', async () => {
    const tree = render(<Badge label="Badge Test" color="red" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
