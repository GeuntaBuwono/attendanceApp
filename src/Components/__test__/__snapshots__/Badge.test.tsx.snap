// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Render Correctly 1`] = `
<View
  className="sc-gsnTZi guFzVf"
  color="red"
>
  <Text
    className="sc-bczRLJ bKLDuK"
    color="#2C2C2B"
    sizeVariant="small"
    style={
      Object {
        "color": "#FFF",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Badge Test
  </Text>
</View>
`;
